### PR TITLE
More tries sync fixes

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed the wrong trie node size estimation when inserting an intercepted trie node in the cache
- made the trie syncer to not depend upon a trie instance

### CODE REVIEW

File: data/trie/sync.go [hunk 1]
```diff
@@ -285,10 +282,11 @@ func (ts *trieSyncer) getNode(hash []byte) (node, error) {
 
 	n, ok := ts.interceptedNodes.Get(hash)
 	if ok {
+		ts.interceptedNodes.Remove(hash)
```
Comment (sasurobert) [inline data/trie/sync.go:285]: the interceptor does not delete the unused infos ?

**Review by BeniaminDrasovean — DISMISSED** [review]

File: data/trie/sync.go (hunk 1 above)
Comment (iulianpascalau) [reply data/trie/sync.go:285]: no, it just stores in the cache what it receives. Just like in the case of transactions, blocks and so on.

File: data/trie/extensionNode.go [hunk 2]
```diff
@@ -643,6 +643,18 @@ func (en *extensionNode) getNextHashAndKey(key []byte) (bool, []byte, []byte) {
 	return false, wantHash, nextKey
 }
 
+func (en *extensionNode) sizeInBytes() int {
+	if en == nil {
+		return 0
+	}
+
+	// hasher + marshalizer + child + dirty flag = 3 * pointerSizeInBytes + 1
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply data/trie/extensionNode.go:651]: this is 3, btw :)

File: data/trie/branchNode.go [hunk 3]
```diff
@@ -790,6 +790,21 @@ func (bn *branchNode) getNextHashAndKey(key []byte) (bool, []byte, []byte) {
 	return false, wantHash, nextKey
 }
 
+func (bn *branchNode) sizeInBytes() int {
+	if bn == nil {
+		return 0
+	}
+
+	// hasher + marshalizer + dirty flag = 2 * pointerSizeInBytes + 1
+	nodeSize := len(bn.hash) + 2*pointerSizeInBytes + 1
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply data/trie/branchNode.go]: done

File: data/trie/extensionNode.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply data/trie/extensionNode.go:651]: done

File: data/trie/sync.go (hunk 1 above)
Comment (iulianpascalau) [reply data/trie/sync.go:285]: done

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**